clippy_lints/src/default_constructed_unit_structs.rs

```diff
@@ -70,15 +71,26 @@ impl LateLintPass<'_> for DefaultConstructedUnitStructs {
             && let var @ ty::VariantDef { ctor: Some((hir::def::CtorKind::Const, _)), .. } = def.non_enum_variant()
             && !var.is_field_list_non_exhaustive()
             && !expr.span.from_expansion() && !qpath.span().from_expansion()
+            // do not suggest replacing an expression by a type name with placeholders
+            && !base.is_suggestable_infer_ty()
         {
-            span_lint_and_sugg(
+            let mut removals = vec![(expr.span.with_lo(qpath.qself_span().hi()), String::new())];
+            if expr.span.with_source_text(cx, |s| dbg!(s).starts_with('<')) == Some(true) {
```

Comment:
Looks like you forgot to remove a `dbg` here (maybe we should run clippy_lints dogfood with `-Dclippy::dbg`)

Reply:
Thanks for catching this! I've submitted https://github.com/rust-lang/rust-clippy/pull/14579 to avoid this in the future.